### ISSUE TITLE
string_to_string does not trigger inside a map function

### ISSUE THREAD

**Issue by MarkusTieger** [rust-lang/rust-clippy#14175]
### Summary

If you try to use String::to_string in a map function on an option, the lint does not trigger.

See the example below

### Lint Name

string_to_string

### Reproducer

I tried this code:

```rust
let variable1 = String::new();
let variable2 = Some(&variable1);
let variable3 = variable2.map(String::to_string);
```

I expected to see this happen:
That `let variable3 = variable2.map(String::to_string);` should be changed to `let variable3 = variable2.cloned();`

Instead, this happened:
nothing

### Version

```text
rustc 1.84.1 (e71f9a9a9 2025-01-27)
binary: rustc
commit-hash: e71f9a9a98b0faf423844bf0ba7438f29dc27d58
commit-date: 2025-01-27
host: x86_64-unknown-linux-gnu
release: 1.84.1
LLVM version: 19.1.5
```

**Comment by xtqqczze** [comment]
Related: https://github.com/rust-lang/rust-clippy/issues/16511
